great_expectations/zep/postgres_datasource.py

```diff
@@ -0,0 +1,127 @@
+# there is no template to actualize. It not None, one can pass in the values to be used in the BatchRequest.
+# For a TableAsset one uses a BatchRequest[TableAssetOptions] to specify the template values.
+
+
+@dataclasses.dataclass(frozen=True)
+class TableAssetSplitter:
+    method_name: str
+    column_name: str
+
+
+@dataclasses.dataclass(frozen=True)
+class TableAssetOptions(BatchRequestOptions):
+    # Depending on the splitter method passed in the possible values one could pass in vary. For example for the
+    # splitter `split_on_year_and_month` the possible kwarg keys are `year` and `month`. However, for the splitter
+    # method `split_on_year_and_month_and_day` the possible kwarg keys are `year`, `month` and `day`. Because of
+    # this the table_splitter_kwargs is a loosely typed dictionary.
+    table_splitter_kwargs: Dict[str, Any]
+
+
+class TableAsset(DataAsset[TableAssetOptions]):
+    def __init__(
+            self,
+            datasource: Datasource,
+            name: str,
+            table_name: str,
+            splitter: Optional[TableAssetSplitter] = None,
+    ):
+        super().__init__(name)
+        self.table_name = table_name
+        self.splitter = splitter
+        self._datasource = datasource
+        self._name = name
+
+    @property
+    def name(self):
+        return self._name
+
+    @property
+    def datasource(self):
+        return self._datasource
+
+    def add_splitter(self, splitter) -> TableAsset:
+        self.splitter = splitter
+        return self
+
+    def get_batch_request(self, options: Optional[TableAssetOptions]=None) -> BatchRequest[TableAssetOptions]:
+        return BatchRequest[TableAssetOptions](
+            datasource_name=self.datasource.name,
+            data_asset_name=self.name,
+            options=options
+        )
+
+
+class PostgresDatasource(metaclass=MetaDatasource):
+    asset_types = [TableAsset]
+
+    def __init__(self, name: str, connection_str: str):
+        self.name = name
+        self.execution_engine = SqlAlchemyExecutionEngine(
+            connection_string=connection_str
+        )
+        self.assets: Dict[str, DataAsset] = {}
+
+    def add_table_asset(self,
+                        name: str,
+                        table_name: str,
+                        splitter: Optional[TableAssetSplitter]=None) -> TableAsset:
+        asset = TableAsset(datasource=self, name=name, table_name=table_name, splitter=splitter)
+        self.assets[name] = asset
+        return asset
+
+    def get_asset(self, name) -> DataAsset:
+        return self.assets[name]
+
+    # When we have multiple types of DataAssets on a datasource, the batch_request argument will be a Union type.
+    # We could pattern match on this type to differentiate. Since we need to support python < 3.10, we can use an
+    # if/else condition instead of match and call a helper function fo each specific type.
```

Comment:
This might be a case for actually using `singledispatch` 😄   🤷

Reply:
Yeah, that is a good thought. I'll update this comment.